src/main/java/net/openhft/chronicle/bytes/Byteable.java

```diff
@@ -27,57 +28,78 @@
 import java.nio.channels.FileLock;
 
 /**
- * This Interface allows a reference to off heap memory to be reassigned.
- * <p></p>
- * A reference to off heap memory is a proxy for some memory which sits outside the heap.
+ * An interface for a reference to off-heap memory, acting as a proxy for memory residing outside the heap.
+ * This allows the reference to be reassigned, facilitating dynamic memory management.
  *
- * @param <B> Bytes type
- * @param <U> Underlying type
+ * @param <B> the BytesStore type
+ * @param <U> the underlying type that the BytesStore manages
  */
 public interface Byteable<B extends BytesStore<B, U>, U> {
     /**
-     * This setter for a data type which points to an underlying ByteStore.
-     *  @param bytesStore the fix point ByteStore
-     * @param offset     the offset within the ByteStore
-     * @param length     the length in the ByteStore
+     * Sets the reference to a data type that points to the underlying ByteStore.
+     *
+     * @param bytesStore the fixed-point ByteStore
+     * @param offset     the offset within the ByteStore, indicating the starting point of the memory section
+     * @param length     the length of the memory section within the ByteStore
+     * @throws ClosedIllegalStateException if it is closed
+     * @throws IllegalArgumentException if the provided arguments are invalid
+     * @throws BufferOverflowException if the new memory section extends beyond the end of the ByteStore
+     * @throws BufferUnderflowException if the new memory section starts before the start of the ByteStore
      */
     void bytesStore(@NotNull BytesStore<B, U> bytesStore, @NonNegative long offset, @NonNegative long length)
-            throws IllegalStateException, IllegalArgumentException, BufferOverflowException, BufferUnderflowException;
```

Comment:
not just java doc code changes as well !

Reply:
ClosedIllegalStateException is the only instance of IllegalStateException that will be changed.
This is a documentation change though not technically a javadoc change.